### PR TITLE
Preparation for heroku hosting

### PR DESCRIPTION
**Pull request by alexlincoln**
All thats missing is the custom buildpack, which will come in the next commit.

### CODE REVIEW

File: server/index.js [hunk 1]
```diff
@@ -50,7 +50,7 @@ app.use(compress())
 
 app.use(function (req, res, next) {
   // Force SSL
-  if (config.isProd && req.protocol !== 'https') {
+  if (config.isProd && req.headers['x-forwarded-proto'] !== 'https') {
```
Comment (feross) [inline server/index.js]: This will break when instant.io is not behind an http proxy. Can you change this to a check for either?

File: config.js [hunk 2]
```diff
@@ -1,6 +1,7 @@
 exports.isProd = process.env.NODE_ENV === 'production'
+exports.isHeroku = process.env.DYNO && process.env.PORT
 exports.host = exports.isProd && '23.239.22.146'
 exports.ports = {
-  http: exports.isProd ? 80 : 9100,
-  https: exports.isProd ? 443 : 9101
+  http: exports.isProd ? process.env.PORT || 80 : 9100,
```
Comment (feross) [inline config.js]: How about this instead?

```json
{
  http: process.env.PORT || (exports.isProd ? 80 : 9100),
  https: process.env.PORT || (exports.isProd ? 443 : 9101)
}
```

**Comment by alexlincoln** [comment]
Alright, looks like almost everything builds and works correctly aside from a missing `rtcConfig`. Here's a test link: https://instant-io-test.herokuapp.com/

**Comment by joshterrill** [comment]
I'm seeing an error that says can't find webrtc config, so it's using the default setting without TURN. Is that normal?

**Comment by feross** [comment]
@joshterrill That's a normal warning. It'll go away if you specify twilio credentials to use for the TURN relay server. This is an optional bit that helps WebRTC to continue working in the case where two peers can't establish a direct connection to each other. If you're deploying an Instant.io mirror, you're free to use my turn server. Send me a message, or open an issue, and I'll whitelist your server to access the endpoint that provides the credentials.

**Pull request closed without merging**